From 57dee6e94e6799198a952a75f4372a4d71718214 Mon Sep 17 00:00:00 2001
From: Varun Kumar Reddy Yaparla <vyaparla@habana.ai>
Date: Mon, 27 Dec 2021 13:37:35 +0200
Subject: [PATCH] [SW-67662] gaudi2: align pll speed to nominal hbm 1.6

This change aligns DCORE#_IF and DCORE#_MESH PLLs to
Nominal, HBM 1.6. This change is required for the PLLs
to run at high speed inorder to speed up sha384 calculation
on boot fit itb to complete faster to avoid LKD timeout
issue in dynamic comms protocol

Change-Id: I7b5c3cb2b2d6c82553d9de7fb1871e004796d79e
---

diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index 095b2aa..c74a024 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -108,8 +108,8 @@
 		.name = "DCORE0_IF",
 		.base = mmDCORE0_XBAR_IF_PLL_CTRL_BASE,
 		.nr = 1, .nf = 64, .od = 2, .nb = 1,	/* 1600 */
-		.div_factor = { 1, 2, 3, 7 },
-		.div_sel = { HL_PLL_DIV_PLL,
+		.div_factor = { 0, 1, 1, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
@@ -118,8 +118,8 @@
 		.name = "DCORE1_IF",
 		.base = mmDCORE1_XBAR_IF_PLL_CTRL_BASE,
 		.nr = 1, .nf = 64, .od = 2, .nb = 1,	/* 1600 */
-		.div_factor = { 1, 2, 3, 7 },
-		.div_sel = { HL_PLL_DIV_PLL,
+		.div_factor = { 0, 1, 1, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
@@ -128,8 +128,8 @@
 		.name = "DCORE2_IF",
 		.base = mmDCORE2_XBAR_IF_PLL_CTRL_BASE,
 		.nr = 1, .nf = 64, .od = 2, .nb = 1,	/* 1600 */
-		.div_factor = { 1, 2, 3, 7 },
-		.div_sel = { HL_PLL_DIV_PLL,
+		.div_factor = { 0, 1, 1, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
@@ -138,8 +138,8 @@
 		.name = "DCORE3_IF",
 		.base = mmDCORE3_XBAR_IF_PLL_CTRL_BASE,
 		.nr = 1, .nf = 64, .od = 2, .nb = 1,	/* 1600 */
-		.div_factor = { 1, 2, 3, 7 },
-		.div_sel = { HL_PLL_DIV_PLL,
+		.div_factor = { 0, 1, 1, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
@@ -148,8 +148,8 @@
 		.name = "DCORE0_MESH",
 		.base = mmDCORE0_XBAR_MESH_PLL_CTRL_BASE,
 		.nr = 1, .nf = 34, .od = 1, .nb = 1,	/* 1700 */
-		.div_factor = { 1, 2, 3, 7 },
-		.div_sel = { HL_PLL_DIV_PLL,
+		.div_factor = { 0, 1, 2, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
@@ -158,8 +158,8 @@
 		.name = "DCORE1_MESH",
 		.base = mmDCORE1_XBAR_MESH_PLL_CTRL_BASE,
 		.nr = 1, .nf = 34, .od = 1, .nb = 1,	/* 1700 */
-		.div_factor = { 1, 2, 3, 7 },
-		.div_sel = { HL_PLL_DIV_PLL,
+		.div_factor = { 0, 1, 2, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
